### ISSUE TITLE
RPy2 behaves strangely inside of Parallel/Delayed

### ISSUE THREAD

**Issue by johnmarktaylor91** [joblib/joblib#762]
Hello,

I'm running a script that uses Parallel/Delayed to run a function in parallel over a number of different arguments, and the function contains RPy2 functions. I am getting this weird behavior where I get an error if I don't import the relevant Rpy2 submodules INSIDE the function I'm parallelizing, and instead just import them outside. Below is a simple example script that reproduced the weird behavior. If I run it, when my_func is called by itself, without wrapping it in Parallel/Delayed, then it runs fine. However, if you wrap it, then you get an error ("undefined symbol: PC"). However, if I uncomment out the import lines inside the function, then it runs fine. Does anyone know why this might be? It seems like the scope of my_func is somehow different from that of the enclosing script when it runs inside parallel/delayed, such that the modules it is referring to aren't the same as that of the enclosing script. 

import multiprocessing as mp
from joblib import Parallel, delayed
import numpy as np
import readline
from rpy2 import robjects
from rpy2.robjects import r, pandas2ri
from rpy2.robjects.packages import importr
import pandas as pd

rstats = importr('stats')
base = importr('base')
pandas2ri.activate()

my_df = pd.DataFrame.from_dict({'x':[1,5,2,2,7,1,4,5],
                                'y':['a','a','a','a','b','b','b','b']})
print(r.summary(r.aov(robjects.Formula('x~y'),data=my_df)))

def my_func():
    #import readline
    #from rpy2 import robjects
    #from rpy2.robjects import r, pandas2ri
    #from rpy2.robjects.packages import importr
    #pandas2ri.activate()

    my_df = pd.DataFrame.from_dict({'x':[1,5,2,2,7,1,4,5],
                                    'y':['a','a','a','a','b','b','b','b']})
    print(r.summary(r.aov(robjects.Formula('x~y'),data=my_df)))
    return 1

my_func()
results = Parallel(n_jobs=num_cores)(delayed(my_func)() for i in range(num_cores))

**Comment by aallahyar** [comment]
Indeed, you can use `rpy2` (to call `R` from `Python`) with `joblib` (for parallel processing) which allows for efficient, parallelized R computations, but this requires caution due to R's single-threaded nature. 

As you already discovered, to prevent crashes, initialize rpy2 inside each parallel worker. This way each worker has a fresh and independent R session.
